database/ffldb/db.go

```diff
@@ -1708,11 +1786,25 @@ func (tx *transaction) writePendingAndCommit() error {
 		// so commonly needed.
 		blockHdr := blockData.bytes[0:blockHdrSize]
 		blockRow := serializeBlockRow(location, blockHdr)
+
 		err = tx.blockIdxBucket.Put(blockData.hash[:], blockRow)
 		if err != nil {
 			rollback()
 			return err
 		}
+
+		blockIdx2Bucket, err := tx.blockIdx2Bucket()
```

Comment:
I would suggest moving this outside of the loop.  Once you've looked it up, you can reuse it inside the loop.